### PR TITLE
refactor: ProphuntESP

### PR DESCRIPTION
**Pull request by MukjepScarlet**
- render with PlacementRenderer

### CODE REVIEW

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleProphuntESP.kt [hunk 1]
```diff
@@ -1,250 +1,93 @@
...
+        rendererBlockUpdates.clearSilently()
+    }
 
-        renderEnvironmentForWorld(matrixStack) {
-            dirty = drawBlocks(this, trackedBlocks, colorMode, fullAlpha, drawOutline) || dirty
+    @Suppress("unused")
+    private val networkHandler = handler<PacketEvent> { event ->
+        when (val packet = event.packet) {
+            is BlockUpdateS2CPacket -> rendererBlockUpdates.addBlock(packet.pos)
+            is ChunkDeltaUpdateS2CPacket -> packet.visitUpdates { pos, _ -> rendererBlockUpdates.addBlock(pos) }
```
Comment (ccetl) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleProphuntESP.kt]: You're aware that the block position in `ChunkDeltaUpdateS2CPacket#visitUpdates` is mutable?

**Comment by ccetl** [comment]
You probably can also render falling blocks with the PlacementRenderer. You'd need to add a method that copies the progress of the old block pos to the current block pos and removes the old block pos (not silent).

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleProphuntESP.kt (hunk 1 above)
Comment (MukjepScarlet) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleProphuntESP.kt]: Oops

**Comment by MukjepScarlet** [comment]
https://github.com/user-attachments/assets/821fb6cd-3222-4fc4-8511-90ba6c0fc670

**Comment by MukjepScarlet** [comment]
The PlacementRenderer is using LinkedHashMap but I have not encountered any ConcurrentModificationException yet, hope it won't appear.

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleProphuntESP.kt [hunk 2]
```diff
@@ -1,250 +1,49 @@
...
-    )
-
-    private val colorMode = choices("ColorMode", 0) {
-        arrayOf(
-            ExpirationColor,
-            GenericStaticColorMode(it, Color4b(255, 179, 72, 150)),
-            GenericRainbowColorMode(it)
-        )
-    }
+    private val defaultColor = Color4b(255, 179, 72, 90)
```
Comment (ccetl) [inline src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleProphuntESP.kt]: Is a variable really needed?

**Review by ccetl — APPROVED** [review]

File: src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleProphuntESP.kt (hunk 2 above)
Comment (MukjepScarlet) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleProphuntESP.kt]: I don't want to create it 2 times

Comment (ccetl) [reply src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/render/ModuleProphuntESP.kt]: I only see one usage.

**Pull request merged**